bundle/src/main/java/com/adobe/acs/commons/wcm/impl/SiteMapServlet.java

```diff
@@ -161,6 +215,44 @@ private void write(Page page, XMLStreamWriter stream, ResourceResolver resolver)
         stream.writeEndElement();
     }
 
+    private void writeAsset(Asset asset, XMLStreamWriter stream, ResourceResolver resolver) throws XMLStreamException {
+        stream.writeStartElement(NS, "url");
+
+
+        String loc = externalizer.externalLink(resolver, externalizerDomain, asset.getPath());
+        writeElement(stream, "loc", loc);
+
+        if (includeLastModified) {
+            long lastModified = asset.getLastModified();
+            if (lastModified > 0) {
+                writeElement(stream, "lastmod", DATE_FORMAT.format(lastModified));
+            }
+        }
+
+        Resource contentResource = asset.adaptTo(Resource.class).getChild("jcr:content");
```

Comment:
I think (but worth doublechecking) that asset.adaptTo(ValueMap.class) would work here and be a bit cleaner.